pyxem/generators/indexation_generator.py

```diff
@@ -122,6 +123,54 @@ def get_library_FT_dict(template_library, shape, fsize):
     return library_FT_dict
 
 
+class RotationIndexationGenerator:
+    """Generates a template-based indexer that also calculates relative
+    rotation of templates.
+
+    Parameters
+    ----------
+    signal : ElectronDiffraction2D
+        The signal of electron diffraction patterns to be indexed.
+    diffraction_library : DiffractionLibrary
+        The library of simulated diffraction patterns for indexation.
+
+    Notes
+    -----
+    To be used with minimal template libraries whereby the first euler
+    angle is 0. It is this angle that is optimized during indexation.
+    """
+    def __init__(self, signal, diffraction_library):
+        self.signal = signal
+        self.library = diffraction_library
+
+    def correlate(self,
+                  n_largest=5,
+                  method="fast_correlation",
+                  include_phases=None,
+                  optimize_direct_beam=False,
+                  chunks="auto",
+                  delta_r=1,
+                  delta_theta=1,
+                  keep=100,
+                  **kwargs,
+                  ):
```

Comment:
docstrings would help the review on this a lot

Reply:
This generator is a stub, I actually don't use it yet since I'm not using the proper result output. I figured I would focus on the functions in utils/indexation_utils